### ISSUE TITLE
Bug: Github actions python 3.8 build does not succeed

### ISSUE THREAD

**Issue by jnnr** [rl-institut/oemof-B3#114]
The 3.8 build runs for a long time and eventually fails. This came up recently on different branches:

* on dev https://github.com/rl-institut/oemof-B3/runs/4935341439?check_suite_focus=true
* on #91 https://github.com/rl-institut/oemof-B3/runs/4935871480?check_suite_focus=true
* on #76 https://github.com/rl-institut/oemof-B3/runs/4920954417?check_suite_focus=true . This one finished after a long time

Seems to be connected to the installation of the frictionless package: **ResolutionTooDeep**. Maybe because it does not find a version that does not raise a dependency conflict? We first suspected that it is connected to the demandlib requirement in #76, but this does not seem to be the cause as it happens on other branches, too.

**Comment by SabineHaas** [comment]
`frictionless` is installed in `oemoflex`. However, `pipdeptree` does not give any dependency warning. Also, when we had dependency issues in other projects, the Error usually told us so. 

Reasons for `ResolutionTooDeep`  seem to be unknown, see  ["2. Resolution too-deep "](https://wiki.python.org/psf/PackagingWG/2020-05-05-pip).

I tried the installation locally --> takes very long, too, I aborted.
So, I tried with a lower pip version: this worked :+1: 
However,  #117, where I installed this lower version of pip did not solve the problem :/

**Comment by khelfen** [comment]
I think this is an issue of `oemoflex` or `oemof.tabular` (or a dependency of those) and you should probably open an Issue there. First I'm getting this warning when running `pip install -e .[dev]`:

```
  WARNING: Missing build requirements in pyproject.toml for oemoflex@ git+https://git@github.com/rl-institut/oemoflex@dev#egg=oemoflex from git+https://****@github.com/rl-institut/oemoflex@dev#egg=oemoflex (from oemof-B3==0.0.0).
  WARNING: The project does not specify a build backend, and pip cannot fall back to setuptools without 'wheel'.
```

First thing to do therefore is to add `"wheel"` and/or a `build-backend` to the `pyproject.toml` of `oemoflex` and build it again. E.g.

```
[build-system]
requires = ["setuptools", "wheel"]
build-backend = "setuptools.build_meta"
```

Second somewhere down the `oemof` installation it seems a dependency or `oemof` itself is not pinning it's dependencies correctly. That's probably why it tries to install so many packages multiple times.

```
INFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter constraints to reduce runtime. See https://pip.pypa.io/warnings/backtracking for guidance. If you want to abort this run, press Ctrl + C.
```

See the [link](https://pip.pypa.io/warnings/backtracking) `pip` provides and [this OS answer for an example on how to resolve this.](https://stackoverflow.com/a/65793484). In the end you need to pin down all conflicted packages to make the installation smooth and prevent the timeout in your CI tests. I logged the installation output from running `pip install -e .[dev]` and attached it so you can check it. I used `python3.8.10`.

BUT after hours of trying it fails to build on my local machine as well with the `ResolutionTooDeep` error. I also tried `python3.9` which has the same problems. `jsonschema` (like Hendrik pointed out) and especially `frictionless` (here it fails on my local machine) take by far the longest to resolve. I tried to pin them down and ran `pip install -e .[dev]` again but this resulted in `pip` having to backtrack other packages, which it didn't have to backtrack before... If you are lucky fixing the first WARNING already resolves this. But if you want to try to resolve it from `oemof-B3` these are the packages which `pip` backtracks (before the `ResolutionTooDeep` error) and would need to be pinned down:

```
marko
jinja2
isodate
importlib-resources
importlib-metadata
idna
gitdb
cligj
click-plugins
charset-normalizer
chardet
certifi
babel
alabaster
wrapt
toposort
tabulate
stopit
ratelimiter
psutil
packaging
click
tableschema
datapackage
oemof-tabular
nbformat
iniconfig
gitpython
jsonschema
frictionless
```

I was able to find a (dirty) workaround for me by forcing order of installation. By running the installation for `oemoflex` first and then the rest, it works for me and the tests are passing. Therefore I changed the `setup.py` as seen in my PR. So after that I can run `pip install -e .` first and than `pip install -e .[full]` without troubles (there is still some backtracking but it's able to resolve it).

**Comment by khelfen** [comment]
On further inspection it seems to be a clash between `oemoflex` and `geopandas`. Without `geopandas` pip install -e .[dev] works as expected.

**Comment by SabineHaas** [comment]
In #101 the python 3.7 build fails now also during installation of dependencies.

**Comment by khelfen** [comment]
@jnnr & @SabineHaas did you see my PR #118? I couldn't add you as reviewers but it should fix your issue for now

**Comment by SabineHaas** [comment]
Thank you @khelfen for this fix! :)

I have tested it locally with python 3.8. 
There I also got a few times the message `INFO: This is taking longer than usual. You might need to provide the dependency resolver with stricter constraints to reduce runtime. See https://pip.pypa.io/warnings/backtracking for guidance. If you want to abort this run, press Ctrl + C.` 
However, oemof-B3 installed in a reasonable time. 

To be able to go on merging PRs with running checks I will merge this PR now but will keep open the issue.

_Originally posted by @SabineHaas in https://github.com/rl-institut/oemof-B3/issues/118#issuecomment-1032424732_
